fugue/dataframe/pandas_dataframe.py

```diff
@@ -103,14 +106,38 @@ def _select_cols(self, cols: List[Any]) -> DataFrame:
             self.native[schema.names], schema, pandas_df_wrapper=True
         )
 
-    def rename(self, columns: Dict[str, str]) -> "DataFrame":
+    def rename(self, columns: Dict[str, str]) -> DataFrame:
         try:
             schema = self.schema.rename(columns)
         except Exception as e:
             raise FugueDataFrameOperationError(e)
         df = self.native.rename(columns=columns)
         return PandasDataFrame(df, schema, pandas_df_wrapper=True)
 
+    def alter_columns(self, columns: Any) -> DataFrame:
+        new_schema = self._get_altered_schema(columns)
+        if new_schema == self.schema:
+            return self
+        new_pdf = pd.DataFrame(self.native.to_dict(orient="series"))
+        for k, v in new_schema.items():
+            if not v.type.equals(self.schema[k].type):
+                old_type = self.schema[k].type
+                new_type = v.type
+                # int -> str
```

Comment:
I was wondering why there are only two cases here (similar with Dask DataFrame) with `int to str` and `bool to str`. I assume these are the cases where NULLs can not be casted but other cases work? Thinking specifically about NULL casting as datetime.

Reply:
there are also datetime and double, NAT and NAN should be casted to NULLS, and these are all in the unit test suite, so i didn't add them here meaning pandas and dask can handle them or in the new DataFrame init, the enforce type can handle them.

Anyway, the tests are their to ensure they all have expected behaviors